component/azstorage/block_blob.go

```diff
@@ -507,9 +507,10 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 		if !dirList[blobInfo.Name] {
 			//log.Info("BlockBlob::List : meta file does not exists for dir %s", blobInfo.Name)
 			// For these dirs we get only the name and no other properties so hardcoding time to current time
+			name := strings.TrimSuffix(blobInfo.Name, "/")
 			attr := &internal.ObjAttr{
-				Path:  split(bb.Config.prefixPath, blobInfo.Name),
-				Name:  filepath.Base(blobInfo.Name),
+				Path:  split(bb.Config.prefixPath, name),
```

Comment:
This is a working code, do we have some bug here ?

Reply:
I think we expect names to not have a trailing / so I am trimming it. Storage returns these dir prefix names with a / at the end.

Reply:
This is to make the test case TestReadDirNoVirtualDirectory get the expected result